Assets/!/Scripts/PlayerAudio.cs

```diff
@@ -1,57 +1,131 @@
+
+    enum RandomClip
+    {
+        Walk, Run, DieScream, Blood, Other
+    }
 
     void Start()
     {
         audioSource = GetComponent<AudioSource>();
+        InitializeAudioClipDictionary();
+    }
+
+    private void InitializeAudioClipDictionary()
+    {
+        audioClipDictionary = new Dictionary<RandomClip, Audio[]>
+        {
+            { RandomClip.Walk, walkAudio },
+            { RandomClip.Run, runAudio },
+            { RandomClip.DieScream, dieScreamAudio },
+            { RandomClip.Blood, bloodAudio },
+            { RandomClip.Other, otherAudio }
+        };
     }
 
     public void Walk()
     {
-        PlayAudio("Walk", 0.8f);
+        PlayRandomAudio(RandomClip.Walk, 0.8f);
+    }
+
+    public void LeftCrouched()
+    {
+        PlayAudio(GetSpecificAudio("LeftCrouch", audioClipDictionary[RandomClip.Other]), 0.5f);
     }
 
-    public void Crouched()
+    public void RightCrouched()
     {
-        PlayAudio("Crouched", 0.5f);
+        PlayAudio(GetSpecificAudio("RightCrouch", audioClipDictionary[RandomClip.Other]), 0.5f);
     }
 
     public void Crawl()
     {
-        PlayAudio("Crawl", 0.20f);
+        PlayAudio(GetSpecificAudio("Crawl", audioClipDictionary[RandomClip.Other]), 0.20f);
     }
 
     public void Run()
     {
-        PlayAudio("Walk", 1f);
+        PlayRandomAudio(RandomClip.Run, 1f);
+    }
+
+    public void DieScream()
+    {
+        PlayRandomAudio(RandomClip.DieScream, 0.8f);
+    }
+
+    public void Blood()
+    {
+        PlayRandomAudio(RandomClip.Blood, 3.5f);
+    }
+
+    public void BulletHit()
+    {
+        PlayAudio(GetSpecificAudio("BulletHit", audioClipDictionary[RandomClip.Other]), 1.5f);
     }
 
     public void Jump()
     {
-        PlayAudio("Jump", 1f);
+        PlayAudio(GetSpecificAudio("Jump", audioClipDictionary[RandomClip.Other]), 1f);
     }
 
-    public void PlayAudio(string clipName, float volume)
+    public void FallDown()
     {
-        Audio audioToPlay = System.Array.Find(audioPlayer, a => a.id == clipName);
+        PlayAudio(GetSpecificAudio("FallDown", audioClipDictionary[RandomClip.Other]), 0.7f);
+    }
+
+    private void PlayRandomAudio(RandomClip clipType, float volume)
+    {
+        Audio audioClip = GetRandomAudioClip(clipType);
+        PlayAudio(audioClip, volume);
+    }
 
-        if (audioToPlay.clip == null)
+    public void PlayAudio(Audio audioClip, float volume)
+    {
+        if (audioClip.clip == null)
         {
-            UnityEngine.Debug.LogWarning("Audio clip not found: " + clipName);
+            UnityEngine.Debug.LogWarning("Audio clip not found: " + audioClip.id);
             return;
         }
 
-
-        audioSource.clip = audioToPlay.clip;
+        audioSource.clip = audioClip.clip;
         audioSource.volume = volume;
         audioSource.Play();
         audioSource.loop = false;
     }
 
+    private Audio GetSpecificAudio(string clipName, Audio[] audioArray)
```

Comment:
Why do you need a specific audio?

When we want to play a sound we choose randomly from the array

Reply:
Some sounds have only one clip, some have more so we choose randomly.In this case, could you give some advice about the different ways

Reply:
Or delete other array,make those with only one clip choice also individual array?

Reply:
If we have only one clip, the random will return always the same clip, no need to have a separate function for this case.
The problem is, what happens if you add one clip? you need to change the code